### PR TITLE
reorganize docs folder

### PR DESCRIPTION
**Pull request by tisonkun**
Following the discussion about Dev Guide https://github.com/engula/engula/discussions/84#discussioncomment-1610368, this PR reorganizes the docs folder into:

```
docs
├── README.md
├── dev
│   ├── README.md
│   ├── api-guidelines.md
│   ├── cli-guidelines.md
│   ├── get-started.md
│   ├── maintainer-guide.md
│   └── release-manual.md
├── rfcs
│   ├── 0000-template.md
│   ├── 0041-engula.md
│   ├── 0054-microunit.md
│   └── README.md
└── user
```

We will continuously collect effective guidelines to Engula development.

Signed-off-by: tison <wander4096@gmail.com>

### CODE REVIEW

**Comment by tisonkun** [comment]
cc @huachaohuang 

If this PR gets merged first, #77 & #80 should be updated correspondingly.

**Review by PsiACE — DISMISSED** [review]
LGTM

**Review by huachaohuang — COMMENTED** [review]
I will need some time to get more understanding about RFCs before reviewing this PR thoroughly.

A quick feedback for now: I think the top-level design documents don't fit in RFCs. The top-level design is intended to lay the groundwork for the project. RFCs are intended to improve the project based on the groundwork. And we may also want to keep the top-level design/architecture up to date as the project evolves. But RFCs are rarely updated once accepted (they can be deprecated thought).

File: docs/rfcs/README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+It is generally a good idea to pursue feedback from other project developers beforehand, to ascertain that the RFC may be desirable; having a consistent impact on the project requires concerted effort toward consensus-building.
+
+The most common preparations for writing and submitting an RFC include talking the idea over on our [official Discord server](https://discord.gg/AN6vgVXaHC) or posting "pre-RFCs" on the [discussion forum](https://github.com/engula/engula/discussions).
+
+## What the process is
+
+In short, to get a major feature added to Engula, one must first get the RFC merged under this directory as a markdown file. At that point the RFC is "active" and may be implemented with the goal of eventual inclusion into Engula.
+
+* Fork the this repository.
+* Copy `0000-template.md` to `1234-my-feature.md` (where "my-feature" is descriptive). The RFC number is the number of "pre-RFC" discussion in discussion forum.
```
Comment (huachaohuang) [inline docs/rfcs/README.md:22]: Hmm, what if we have a discussion number larger than 10000? It doesn't seem too hard to get there.

**Comment by huachaohuang** [comment]
I like how [cockroachdb](https://github.com/cockroachdb/cockroach/tree/master/docs) organizes `docs`. It can be simplified as follows:

```
docs/
  README.md
  some top-level design documents...
  dev/
  rfcs/
  images/
```

The top-level design documents define common concepts, terms, and modules, so they are the prerequisites for `dev/` and `rfcs/`. `dev/` describes the internals, workflows, project organization, etc.

**Comment by huachaohuang** [comment]
As for user documents, we may want to put them to another repo (for example, the website repo) because contributors of user documents are very different people, they don't have to understand how Engula works.

**Comment by tisonkun** [comment]
@huachaohuang thanks for your input. I agree to follow the ideas. Updating...

**Comment by tisonkun** [comment]
@huachaohuang updated. 

For the user docs, shall we put them into engula.github.io repo when we actually collect/generate some?

File: docs/rfcs/README.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+It is generally a good idea to pursue feedback from other project developers beforehand, to ascertain that the RFC may be desirable; having a consistent impact on the project requires concerted effort toward consensus-building.
+
+The most common preparations for writing and submitting an RFC include talking the idea over on our [official Discord server](https://discord.gg/AN6vgVXaHC) or posting "pre-RFCs" on the [discussion forum](https://github.com/engula/engula/discussions).
+
+## What the process is
+
+In short, to get a major feature added to Engula, one must first get the RFC merged under this directory as a markdown file. At that point the RFC is "active" and may be implemented with the goal of eventual inclusion into Engula.
+
+* Fork the this repository.
+* Copy `00000000-template.md` to `YYYYMMDD-my-feature.md` (where "my-feature" is descriptive).
```
Comment (tisonkun) [inline docs/rfcs/README.md:22]: I just notice that with `my-feature` description even use a `YYYYMMDD` format, filename won't conflict. And when contributors talk about the feature, they almost use the descriptive name, which is better than a number. (When one uses the number, he should always make an addition "a.k.a. the blablabla feature".)

File: docs/rfcs/00000000-template.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
+---
+status: '[accepted|released|discarded]'
```
Comment (huachaohuang) [inline docs/rfcs/00000000-template.md:2]: We need a process for these stages but we can postpone it for now.

**Review by huachaohuang — APPROVED** [review]
LGTM, thanks for helping with this!

File: docs/rfcs/00000000-template.md (hunk 3 above)
Comment (tisonkun) [reply docs/rfcs/00000000-template.md:2]: Yes. Let's conclude from real-world example. I think most potential RFCs in the near feature can change status with a consensus in our community.

**Pull request merged**